scripts/generate_samples_list.dart

```diff
@@ -0,0 +1,63 @@
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+/// Run from arcgis-maps-sdk-flutter-samples root directory
+/// dart run scripts/generate_samples_list.dart
+void main() {
+  generateSamplesList();
+}
+
+/// Compiles all of the individual sample README.metadata.json files into a single JSON asset to feed into the Sample Viewer application.
+/// The output is in assets/generated_samples_list.json
+void generateSamplesList() {
+  Map<String, dynamic> samples = {};
+
+  print('Generating samples list....');
+
+  // get the lib/samples directory
+  final currentDirectory = Directory.current;
+  final samplesDirectory = Directory('${currentDirectory.path}/lib/samples');
+
+  if (samplesDirectory.existsSync()) {
+    // get all the individual sample directories
+    final individualSampleDirectories = samplesDirectory.listSync();
+
+    // iterate through each directory and get the README.metadata.json file for each sample
+    for (final directory in individualSampleDirectories) {
+      final directoryName = directory.path.split('/').last;
+      final metadataFile = (directory as Directory)
+          .listSync()
+          .firstWhere((file) => file.path.contains('metadata.json')) as File;
+      // get the json string for the sample
+      Map<String, dynamic> sampleJsonContent =
```

Comment:
`final sampleJsonContent`

Reply:
Will leave per the above comment - since jsonDecode returns `dynamic` 👍

Reply:
nevermind about this one, per other discussion